File: csharp/Platform.Interfaces.Tests/InterfacesTests.cs
using System.IO;
using System.Reflection.PortableExecutable;
using Xunit;

namespace Platform.Interfaces.Tests
{
    public static class InterfacesTests
    {
        [Fact]
        public static void BuildTest()
        {
            ICounter<int, int>? c1 = null;
            ICounter<int>? c2 = null;
            ICriterionMatcher<int>? cm1 = null;
            IFactory<int>? f1 = null;
            IProperties<int, int, int>? p1 = null;
            IProperty<int, int>? p2 = null;
            IProvider<int, int>? p3 = null;
            IProvider<int>? p4 = null;
            ISetter<int, int>? s1 = null;
            ISetter<int>? s2 = null;

            Assert.Null(c1);
            Assert.Null(c2);
            Assert.Null(cm1);
            Assert.Null(f1);
            Assert.Null(p1);
            Assert.Null(p2);
            Assert.Null(p3);
            Assert.Null(p4);
            Assert.Null(s1);
            Assert.Null(s2);
        }

        [Fact]
        public static void AssemblyContainsEmbeddedPortablePdb()
        {
            using var assembly = File.OpenRead(typeof(ICounter<>).Assembly.Location);
            using var peReader = new PEReader(assembly);

            Assert.Contains(
                peReader.ReadDebugDirectory(),
                entry => entry.Type == DebugDirectoryEntryType.EmbeddedPortablePdb);
        }
    }
}

File: cpp/Platform.Interfaces.Tests/InterfacesTests.cpp
namespace Platform::Interfaces::Tests
{
    TEST_CLASS(Interfaces)
    {
        public: TEST_METHOD(BuildTest)
        {
            ICounter<std::int32_t, std::int32_t> c1 = nullptr;
            ICounter<std::int32_t> c2 = nullptr;
            ICriterionMatcher<std::int32_t> cm1 = nullptr;
            IFactory<std::int32_t> f1 = nullptr;
            IProperties<std::int32_t, std::int32_t, std::int32_t> p1 = nullptr;
            IProperty<std::int32_t, std::int32_t> p2 = nullptr;
            IProvider<std::int32_t, std::int32_t> p3 = nullptr;
            IProvider<std::int32_t> p4 = nullptr;
            ISetter<std::int32_t, std::int32_t> s1 = nullptr;
            ISetter<std::int32_t> s2 = nullptr;
        }
    };
}
